src/bindgen/library.rs

```diff
@@ -443,4 +450,53 @@ impl Library {
             x.mangle_paths(&monomorphs);
         }
     }
+
+    // Apply associated type replacement to everything
+    fn replace_assoc_types(&mut self){
+
+        let assoc_map = &self.assoc_types;
+
+        // Replace associated types in function signatures
+        for func in self.functions.iter_mut(){
+            // Check return type
+            if let Some(ty) = find_concrete(&func.ret, assoc_map){
+                func.ret = ty
+            };
+
+            // Check arguments type
+            for arg in func.args.iter_mut(){
+                if let Some(ty) = find_concrete(&arg.ty, assoc_map){
+                    arg.ty = ty
+                };
+            }
+        }
+
+        // Replace associated types in structs
+        self.structs.for_all_items_mut(|struct_| {
+            for field in struct_.fields.iter_mut(){
+                if let Some(ty) = find_concrete(&field.ty, assoc_map){
+                    field.ty = ty;
+                }
+            }
+        });
+
+        // Replace associated types in constants
+        self.constants.for_all_items_mut(|const_|{
+            if let Some(ty) = find_concrete(&const_.ty, assoc_map){
+                const_.ty = ty;
+            }
+        });
+    
```

Comment:
you didn't cover type aliases, they can refer to associated type like this:

```
type Kita = <u32 as Trait>::Assoc;
```

also cover with tests

Reply:
you didn't cover unions, they can refer to associated type like this:

```
union Kita {
    a: <u32 as Trait>::Assoc,
    b: u32
}
```
also cover with tests

Reply:
and what about enums such as:
```rs
enum Kara {
    A(<u32 as Kita>::Assoc),
    B { x: <u32 as Kita>::Assoc },
}
```

is this supported?

Reply:
also `static`s

Reply:
basically, all fields in `Library` have a `Type` inside except for `config` and `source_files`